.circleci/config.yml

```diff
@@ -3,7 +3,7 @@ jobs:
   test:
     working_directory: /go/src/github.com/Azure/acs-engine
     docker:
-      - image: microsoft/acs-engine:latest
```

Comment:
The go-dev image is a development image that contains things like ginkgo, az cli, helm, etc... Its merely for CI and development use. Having an image like `microsoft/acs-engine` as part of our development environment is confusing since that should be a distribution image.